### DISCUSSION TITLE
install from location other than pypi.org

### DISCUSSION THREAD

**Discussion by nschloe** [tox-dev/tox#2410] [Q&A] [answered]
In my tox tests, I would like to install pyvista from
```
pip install --find-links https://wheels.pyvista.org/ pyvista
```
(This is necessary because it provides a native VTK Python 3.10 wheel which VTK upstream doesn't yet.)

How can I do this in my `tox.ini`?

**Comment by jugmac00** [comment]
Is pyvista your only external dependency? I assume not 😉

**Reply by nschloe** [reply]
You're right, it's not.

**Comment by jugmac00** [comment, marked as answer]
It is not like I needed that before but it should work like this:

- create a dedicated requirements.txt file for pyvista only
- add the index url on the first line, see https://stackoverflow.com/a/2477610/672833
- add pyvista on the second line
- in your deps section add one line as following `-r requirements.txt`

Untested, but this should do.

Alternatively you could use `commands_pre` to install the single dependency via `python -m pip install...`.

I am not near my laptop so both suggestions are untested.

Please report back whether the one or the other solution worked. If not I'll try find a solution tomorrow.

**Reply by nschloe** [reply]
I'm using `commands_pre`, thanks for the tip!
